templates/admin/user/list.tmpl

```diff
@@ -10,7 +10,66 @@
+							<div class="ui divider"></div>
+							<label class="item"><input type="radio" name="status_filter[is_restricted]" value="0"> {{.i18n.Tr "admin.users.list_status_filter.not_restricted"}}</label>
+							<label class="item"><input type="radio" name="status_filter[is_restricted]" value="1"> {{.i18n.Tr "admin.users.list_status_filter.is_restricted"}}</label>
+							<div class="ui divider"></div>
+							<label class="item"><input type="radio" name="status_filter[is_prohibit_login]" value="0"> {{.i18n.Tr "admin.users.list_status_filter.not_prohibit_login"}}</label>
+							<label class="item"><input type="radio" name="status_filter[is_prohibit_login]" value="1"> {{.i18n.Tr "admin.users.list_status_filter.is_prohibit_login"}}</label>
+							<div class="ui divider"></div>
+							<label class="item"><input type="radio" name="status_filter[is_2fa_enabled]" value="1"> {{.i18n.Tr "admin.users.list_status_filter.is_2fa_enabled"}}</label>
+							<label class="item"><input type="radio" name="status_filter[is_2fa_enabled]" value="0"> {{.i18n.Tr "admin.users.list_status_filter.not_2fa_enabled"}}</label>
+						</div>
+					</div>
+
+					<!-- Sort Menu Item -->
+					<div class="ui dropdown type jump item">
+						<span class="text">
+							{{.i18n.Tr "repo.issues.filter_sort"}} {{svg "octicon-triangle-down" 14 "dropdown icon"}}
+						</span>
+						<div class="menu">
+							<button class="item" name="sort" value="oldest">{{.i18n.Tr "repo.issues.filter_sort.oldest"}}</button>
+							<button class="item" name="sort" value="newest">{{.i18n.Tr "repo.issues.filter_sort.latest"}}</button>
+							<button class="item" name="sort" value="alphabetically">{{.i18n.Tr "repo.issues.label.filter_sort.alphabetically"}}</button>
+							<button class="item" name="sort" value="reversealphabetically">{{.i18n.Tr "repo.issues.label.filter_sort.reverse_alphabetically"}}</button>
+							<button class="item" name="sort" value="recentupdate">{{.i18n.Tr "repo.issues.filter_sort.recentupdate"}}</button>
+							<button class="item" name="sort" value="leastupdate">{{.i18n.Tr "repo.issues.filter_sort.leastupdate"}}</button>
+						</div>
+					</div>
+				</div>
+
+				<!-- Search Text -->
+				<div class="ui fluid action input" style="max-width: 70%;">
+					<input name="q" value="{{.Keyword}}" placeholder="{{.i18n.Tr "explore.search"}}..." autofocus>
+					<button class="ui blue button">{{.i18n.Tr "explore.search"}}</button>
+				</div>
+
+				{{/* here we have valid go template syntax, but eslint doesn't like it and reports "error  Parsing error: Unexpected token {" */}}
+				<script>
+					<!-- /* eslint-disable */ -->
+					(function() {
+						window.config.PageData.adminUserListSearchForm = {
+							statusFilterMap: {{.StatusFilterMap}},
+							sortType: {{.SortType}} || 'oldest'
+						}
+					})();
+				</script>
```

Comment:
please put this js in a file linked to index.js

Reply:
`StatusFilterMap` is a JSON object, it can not be easily put into HTML `data`.

We must have a clear method to unify how to pass complex data to JavaScript code. If we do not introduce `PageData`, there will be more messy frontend codes in future.

For example, the legacy code `var ActivityTopAuthors = {{Json .ActivityTopAuthors | SafeJS}};` in `templates/repo/activity.tmpl` should also be refactored to similar mechanism. I'd like to make a refactor for legacy code if this PR is accepted.

Reply:
Another problem for passing template variables in HTML `data` is: different JavaScript modules can not share context data. 

For example, in the `view issue` page, there is no stable way to get an issue id or index. All modules in this page use their own method. When I want to write something to save and view content history, I have to introduce new variables. 

If there is something like `PageData.currentIssueId`, `PageData.currentIssueIndex` and `PageData.currentIssueUrl`, many things will become simple.